### PR TITLE
Reformat and reorganize proxy configuration

### PR DESCRIPTION
**Pull request by glasser**
- Generally describe things as JSON instead of protobuf. "array of" rather than
  "repeated", no references to "oneof", etc.

- Drop 'Config.' from all message names and rename the top one to "Top level
  options"

- Switch to H2s so that the sidebar TOC works

- Switch table to DL, which looks better when description lengths vary.

- Clean up some other wording.

### CODE REVIEW

File: source/proto-doc.md [hunk 1]
```diff
@@ -1,249 +1,315 @@
...
+<dd>The AWS secret access key for the AWS IAM access key ID specified in the `awsAccessKeyId` field.</dd>
 
+</dl>
 
 
 
 <a name="mdg.engine.config.proto.Config.PersistedQueries"/>
 
-### Config.PersistedQueries
+## PersistedQueries
```
Comment (daniman) [inline source/proto-doc.md]: Is this called PersistedQueries and not PersistedQueriesCache for a reason?

Comment (glasser) [reply source/proto-doc.md]: It's configuration of the general persisted query feature. The fact that it uses a cache as part of its implementation seems just like implementation details to me.  eg if we had a PQ feature that used the same protocol but you loaded in all of your PQs ahead of time (in configuration or from Engine servers) then there wouldn't really be a "cache" at all.

File: source/proto-doc.md [hunk 2]
```diff
@@ -1,249 +1,315 @@
...
+<a name="mdg.engine.config.proto.Config"/>
 
-Note that proto3 JSON flattens "oneof" messages, such that: `oneof foobar { bool foo = 1; bool bar = 2; }` is represented as: `{ "foo": true }`.
+## Top level options
+These are the top-level options for the `new ApolloEngine()` or `new ApolloEngineLauncher()` constructors or for the `engine-config.json` file.
 
 
-| Field | Type | Description |
-| ----- | ---- | ----------- |
-| apiKey |  string | API key for the service. Get this from your [Engine home](https://engine.apollographql.com). This field is required. |
```
[the remark this answers is not in the document]
Comment (glasser) [reply source/proto-doc.md]: Done

[the remark this answers is not in the document]
Comment (glasser) [reply source/proto-doc.md]: Done (at end)

File: source/proto-doc.md [hunk 3]
```diff
@@ -1,249 +1,315 @@
...
-| origins | repeated  [Config.Origin](#mdg.engine.config.proto.Config.Origin)  | Origins represent the GraphQL servers to which the Proxy will send requests. If you're using the `ApolloEngine` class from the `apollo-engine` npm package, you don't need to specify origins: the package will generate one automatically for you, or will fill in the url automatically if you have configured other origin properties. |
```
[the remark this answers is not in the document]
Comment (glasser) [reply source/proto-doc.md]: Done

File: source/proto-doc.md [hunk 4]
```diff
@@ -1,249 +1,315 @@
...
-| frontends | repeated  [Config.Frontend](#mdg.engine.config.proto.Config.Frontend)  | The list of frontends to listen to for GraphQL queries. If not specified, a frontend will be created with default values. |
```
[the remark this answers is not in the document]
Comment (glasser) [reply source/proto-doc.md]: Done

File: source/proto-doc.md [hunk 5]
```diff
@@ -1,249 +1,315 @@
...
-| stores | repeated  [Config.Store](#mdg.engine.config.proto.Config.Store)  | The list of configured stores to cache responses to. |
-| sessionAuth |   [Config.SessionAuth](#mdg.engine.config.proto.Config.SessionAuth)  | The session authorization configuration to use for per-session caching. |
-| logging |   [Config.Logging](#mdg.engine.config.proto.Config.Logging)  | The logging configuration to use. |
-| reporting |   [Config.Reporting](#mdg.engine.config.proto.Config.Reporting)  | The reporting configuration to use. |
-| queryCache |   [Config.QueryResponseCache](#mdg.engine.config.proto.Config.QueryResponseCache)  | The query response cache configuration to use. |
```
[the remark this answers is not in the document]
Comment (glasser) [reply source/proto-doc.md]: changed to

	// Configuration for [caching responses to GraphQL
	// queries](./caching.html). Only responses annotated with the cache-control
	// extension are cached.

**Pull request merged**